### ISSUE TITLE
How to secure files with IntraACL?

### ISSUE THREAD

**Issue by BuFann** [mediawiki4intranet/IntraACL#69]
IntraACL is a powerful security extension, no doubt. And thanks a lot for the great work you have done.

So, is there any feasible way to secure files (including images, pdfs, etc.) when somebody tries to directly access them? I know I can add ACL to the file namespace, but it still can't secure the direct access.

Thank you guys!

**Comment by vitalif** [comment]
Hi, in fact it.s simple: use mediawiki's img_auth.php (google it) and disable direct file access in your apache/nginx config :-)

**Comment by BuFann** [comment]
Thank you, vitalif.
It works fine to protect my files from guests, but still I can't figure out a way to define different access groups under the defination of IntraACL.

**Comment by vitalif** [comment]
Just protect File: pages like normal wiki pages, it should work

**Comment by BuFann** [comment]
Yes it works in some way, and the File: pages can be secured. But what if a
user is trying to access the file directly if he knows the file's link?

Consider this condition: a user in Development Department has the access of
downloading a secret technical file, while a user in Financial Department
doesn't. What if the user in Development Department occasionally sent the
link of the file to the user in Financial Department? The file would be
leaked. So the only way to protect a file is to protect its link. It's not
safe enough as far as I concern, but it should work in most conditions.

Thanks a lot!

2017-01-11 15:50 GMT+08:00 Vitaliy Filippov <notifications@github.com>:

**Comment by vitalif** [comment]
If the only way to access the file is through img_auth.php it will check read permissions for file page each time the user tries to download it directly

**Comment by BuFann** [comment]
YES!
That's so clever! This solved my question!

Thanks! I have to say that IntraACL is really powerful and is the best
access control extention for MediaWiki ever.

2017-01-11 17:00 GMT+08:00 Vitaliy Filippov <notifications@github.com>:

**Comment by BuFann** [comment]
I also think that the issue https://github.com/mediawiki4intranet/IntraACL/issues/55 describes a very similiar situation as I imagined.
I think this issue can also help him.
